cloud/pkg/devicecontroller/controller/upstream.go

```diff
@@ -159,9 +159,9 @@ func (uc *UpstreamController) updateDeviceStatus() {
 				klog.Errorf("Failed to marshal device status %v", deviceStatus)
 				continue
 			}
-			result := uc.crdClient.Patch(MergePatchType).Namespace(cacheDevice.Namespace).Resource(ResourceTypeDevices).Name(deviceID).Body(body).Do(context.Background())
-			if result.Error() != nil {
-				klog.Errorf("Failed to patch device status %v of device %v in namespace %v", deviceStatus, deviceID, cacheDevice.Namespace)
+			err = uc.crdClient.DevicesV1alpha2().RESTClient().Patch(MergePatchType).Namespace(cacheDevice.Namespace).Resource(ResourceTypeDevices).Name(deviceID).Body(body).Do(context.Background()).Error()
+			if err != nil {
+				klog.Errorf("Failed to patch device status %v of device %v in namespace %v, err: %v", deviceStatus, deviceID, cacheDevice.Namespace, err)
```

Comment:
Now we can't use updatestatus to patch the twin info in device status and only can use the restclient, because:

1. Now the device crd yaml don't  set the status subresource, so updatestatus can't work.
2. If we set status subresource in device crd, when we create the device obj, the status section will not be applied in apiserver/etcd, must use updatestatus to add it.  Now we put desired value in twin of status, we need to create obj with status.  In future, maybe **it's need to split the  desired value in twin to spec and leave the reported value in status**.

cc @luogangyi @sailorvii @kevin-wangzefeng

Reply:
Refer to https://book-v1.book.kubebuilder.io/basics/status_subresource.html, we can upgrade CRD  from v1beta1 to v1 to enable status subresource as #2408